### ISSUE TITLE
[Typescript] Schema `ref` property type declaration issue

### ISSUE THREAD

**Issue by Jule-** [Automattic/mongoose#13672]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.6.0

### Node.js version

18.15.0

### MongoDB server version

5.0.11

### Typescript version (if applicable)

5.1.6

### Description

There is an issue when defining cross-database ref in a Schema where the Model has TVirtuals type provided.

```
error TS2322: Type '{ type: typeof Schema.Types.ObjectId; ref: ModelA; }' is not assignable to type 'SchemaDefinitionProperty<ObjectId> | undefined'.
  Types of property 'ref' are incompatible.
    Type 'ModelA' is not assignable to type 'string | Model<any, {}, {}, {}, any> | ((this: any, doc: any) => string | Model<any, {}, {}, {}, any>) | undefined'.
      Type 'Model<IModelA, object, object, IModelAVirtuals, any>' is not assignable to type 'Model<any, {}, {}, {}, any>'.
        Property 'virtualProperty' is missing in type '{}' but required in type 'IModelAVirtuals'.
```

### Steps to Reproduce

```ts
// repro.ts
import { model, Model, Schema, Types } from "mongoose";

interface IModelA {
  name: string;
}

interface IModelAVirtuals {
  readonly virtualProperty: string;
}

type ModelA = Model<IModelA, object, object, IModelAVirtuals>;

const modelASchema = new Schema<IModelA, ModelA, object, object, IModelAVirtuals>({
  name: String,
});

modelASchema.virtual("virtualProperty").get(function () {
  return "Virtual!";
});

const A = model<IModelA, ModelA>("ModelA", modelASchema);

interface IModelB {
  a: Types.ObjectId;
}

const modelBSchema = new Schema<IModelB>({
  // Here we use the Model instance for cross-database population
  // https://mongoosejs.com/docs/populate.html#cross-db-populate
  a: { type: Schema.Types.ObjectId, ref: A },
});

const B = model<IModelB>("ModelB", modelBSchema);
```

```sh
npx ts-node repro.ts
```

### Expected Behavior

The `ref` type declaration should allow virtuals to be present in the Model type.

I think modifying these 2 lines should be enough:

https://github.com/Automattic/mongoose/blob/255493b51db27e61bd84ef9bfef47a0ac4068183/types/schematypes.d.ts#L88

https://github.com/Automattic/mongoose/blob/255493b51db27e61bd84ef9bfef47a0ac4068183/types/schematypes.d.ts#L251

Not sure if this type could be inferred somewhere or if it should only be `Model<any, any, any, any>`.

**Comment by Jule-** [comment]
@IslandRhythms oh yes you are right! So ignore my comments about a potential fix then, it should exist a better solution.

**Comment by vkarpov15** [comment]
This was fixed in 6.9.3, likely with #13066. Upgrade to >= 6.9.3.
